okhttp/src/main/java/io/grpc/okhttp/OkHttpClientStream.java

```diff
@@ -75,7 +76,8 @@
       StatsTraceContext statsTraceCtx,
       TransportTracer transportTracer,
       CallOptions callOptions,
-      boolean useGetForSafeMethods) {
+      boolean useGetForSafeMethods,
+      boolean usePlaintext) {
```

Comment:
I wonder if we need to pass this if this can be deduced from the `transport` argument (of type `OkHttpClientTransport`). For plaintext is it possible that `sslSocketFactory` is null, so all we need is to expose that in `OkHttpClientTransport`. Trying to see if we can avoid redundancy.

Reply:
What we want is just to know if we are using plaintext. In my opinion, passing a boolean is much safer than expose the `sslSocketFactory`.

Reply:
Not exposing `sslSocketFactory`; just a boolean method returning true if `sslSocketFactory == null` (assuming that flow works)

Reply:
That's a great point, changed my implementation.

Reply:
Great, the test `io.grpc.okhttp.OkHttpChannelBuilderTest#usePlaintextCreatesNullSocketFactory' confirms the assumption made here. LGTM